### PR TITLE
[CMAKE] Do not build stressHistFactory if GSL is not found

### PR DESCRIPTION
**Pull request by bluehood**
...because when GSL is not found, HistFactory is not built.

Without this change, when GSL is not present ROOT builds fail with the following error:

```
In file included from /home/eguiraud/ROOT/root/test/stressHistFactory_tests.cxx:25:0,
                 from /home/eguiraud/ROOT/root/test/stressHistFactory.cxx:26:
/home/eguiraud/ROOT/root/test/stressHistFactory_models.cxx:8:46: fatal error: RooStats/HistFactory/Measurement.h: No such file or directory
 #include "RooStats/HistFactory/Measurement.h"
                                              ^
compilation terminated.
test/CMakeFiles/stressHistFactory.dir/build.make:62: recipe for target 'test/CMakeFiles/stressHistFactory.dir/stressHistFactory.cxx.o' failed
```

@hageboeck this fixes the issue I was complaining about today :smile: 

@amadio maybe you can think of a nicer fix, this PR is most of all a bug report.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -212,7 +212,7 @@ endif()
 
 #--stressHistFactory--------------------------------------------------------------------------------
 if(ROOT_roofit_FOUND)
-if(ROOT_xml_FOUND)
+if((xml OR ROOT_xml_FOUND) AND GSL_FOUND) # must use same check as in roofit/CMakeLists.txt
```
Comment (amadio) [inline test/CMakeLists.txt]: Why did you need the `xml OR` part? I think this should be `if(ROOT_xml_FOUND AND ROOT_mathmore_FOUND)`.

Comment (bluehood) [reply test/CMakeLists.txt]: I don't know, I just copied the condition that disables `HistFactory` verbatim

Comment (amadio) [reply test/CMakeLists.txt]: It seems `HistFactory` depends on `XMLParser` only, while `hist2workspace` depends on GSL as well. We need to fix this in RooFit. @hageboeck Does `hist2workspace` depend on `mathmore` or just `GSL`, do you know?

Comment (amadio) [reply test/CMakeLists.txt]: In summary, the `add_directory(histfactory)` should not be conditional, and the dependencies should be handled in `roofit/histfactory/CMakeLists.txt` for `xml` and `GSL`.

**Comment by bluehood** [comment]
Ok it looks like this is not the proper fix :sweat_smile: 
Good luck!
